hack/api-docs/Dockerfile

```diff
@@ -11,20 +11,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM alpine:3.18@sha256:34871e7290500828b39e22294660bee86d966bc0017544e848dd9a255cdf59e0
+FROM alpine:3.19@sha256:51b67269f354137895d43f3b3d810bfacd3945438e94dc5ac55fdac340352f48
 COPY requirements.txt /
 RUN apk add -U --no-cache \
     python3 \
     python3-dev \
-    py3-pip==23.1.2-r0 \
+    py3-pip \
     musl-dev \
     git \
     openssh \
     git-fast-import \
     bash \
     gcc \
     diffutils
-RUN pip3 install -r /requirements.txt
+RUN pip3 install --break-system-packages -r /requirements.txt
```

Comment:
Errrr.... are you sure about this? It might end up with a broken installation and weird errors.

Reply:
[This article](https://pythonspeed.com/articles/externally-managed-environment-pep-668/) sums it up quite nicely.

tl;dr

With the alpine upgrade the python base installation is marked as "externally managed" and one can not install stuff with `pip install <foo>` any more. 
The docker build fails with the following error:

```
$ make image
# ...
 => ERROR [4/5] RUN pip3 install -r /requirements.txt                                                                            1.9s
------                                                                                                                                
 > [4/5] RUN pip3 install -r /requirements.txt:                                                                                       
1.657 error: externally-managed-environment                                                                                           
1.657                                                                                                                                 
1.657 × This environment is externally managed                                                                                        
1.657 ╰─> 
1.657     The system-wide python installation should be maintained using the system
1.657     package manager (apk) only.
1.657     
1.657     If the package in question is not packaged already (and hence installable via
1.657     "apk add py3-somepackage"), please consider installing it inside a virtual
1.657     environment, e.g.:
1.657     
1.657     python3 -m venv /path/to/venv
1.657     . /path/to/venv/bin/activate
1.657     pip install mypackage
1.657     
1.657     To exit the virtual environment, run:
1.657     
1.657     deactivate
1.657     
1.657     The virtual environment is not deleted, and can be re-entered by re-sourcing
1.657     the activate file.
1.657     
1.657     To automatically manage virtual environments, consider using pipx (from the
1.657     pipx package).
1.657 
1.657 note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
1.657 hint: See PEP 668 for the detailed specification.
```

I'm not enough of a pythoniast to make a good call here. I wanted to "make it work" without relying on system package manager (then we rely too much on the distro packaging and their security fix & rollout procedures). Having another tool (venv / pipx) or layer of indirection adds more stuff.

After trying out `pipx` i think `venv` is probably the best solution here. Thanks for flagging that up!

Reply:
Nice! Yeah. Well done making that work better. 😊😊